src/test/java/seedu/address/logic/commands/CopyCommandTest.java

```diff
@@ -0,0 +1,152 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.commands.CommandTestUtil.assertParseFailure;
+import static seedu.address.logic.commands.CopyCommand.MESSAGE_USAGE;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.person.Person;
+import seedu.address.testutil.PersonBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for CopyCommand.
+ */
+public class CopyCommandTest {
+
+    private static final String VALID_INDEX = "1";
+    private static final String INVALID_INDEX = "a";
+    private final Model model = new ModelManager();
+
+    @Test
+    public void execute_validCopy1_success() throws CommandException {
+        Person personToCopy = new PersonBuilder().withTags("Bob").withAssets("Aircon").build();
+        model.addPerson(personToCopy);
+        Index index = Index.fromOneBased(1);
+        boolean[] info = {true, true, true, true, true, true};
+        CopyCommand copyCommand = new CopyCommand(index, info);
+
+        String copiedDetails = "Name: Amy Bee; "
+            + "Phone: 85355255; "
+            + "Email: amy@gmail.com; "
+            + "Address: 123, Jurong West Ave 6, #08-111; "
+            + "Tags: [[Bob]]; "
+            + "Assets: [[Aircon]]";
+
+        String expectedMessage = String.format(Messages.MESSAGE_COPIED, copiedDetails);
+        assertEquals(expectedMessage, copyCommand.execute(model));
+    }
+
+    @Test
+    public void execute_validCopy2_success() throws CommandException {
+        Person personToCopy = new PersonBuilder().withTags("Bob").withAssets("Aircon").build();
+        model.addPerson(personToCopy);
+        Index index = Index.fromOneBased(1);
+        boolean[] info = {false, true, false, true, true, true};
+        CopyCommand copyCommand = new CopyCommand(index, info);
+
+        String copiedDetails = "Phone: 85355255; "
+            + "Address: 123, Jurong West Ave 6, #08-111; "
+            + "Tags: [[Bob]]; "
+            + "Assets: [[Aircon]]";
```

Comment:
Would it be better to reuse the variables from `PersonBuilder.java`?

Reply:
Uses variables from `PersonBuilder.java`